server/sonar-server/src/main/java/org/sonar/server/ui/ViewProxy.java

```diff
@@ -272,6 +274,24 @@ public boolean acceptsAvailableMeasures(String[] availableMeasures) {
     }
   }
 
+  public boolean isUserAuthorized() {
+    boolean authorized = userRoles.length == 0;
+    for (String userRole : getUserRoles()) {
+      authorized |= UserSession.get().hasGlobalPermission(userRole);
+    }
+    return authorized;
+  }
+
+  public boolean isUserAuthorized(ComponentDto component) {
+    boolean authorized = userRoles.length == 0;
+    for (String userRole : getUserRoles()) {
+      authorized |= (UserRole.VIEWER.equals(userRole)
+        || UserRole.USER.equals(userRole)
+        || UserSession.get().hasProjectPermissionByUuid(userRole, component.uuid()));
```

Comment:
Links to unauthorized pages must not be displayed to end-users. So the condition must check only the expected permission, not UserRole.VIEWER and USER.